src/GTP.cpp

```diff
@@ -516,15 +516,27 @@ bool GTP::execute(GameState & game, std::string xinput) {
     } else if (command.find("heatmap") == 0) {
         std::istringstream cmdstream(command);
         std::string tmp;
-        int rotation;
+        std::string rotation;
 
         cmdstream >> tmp;   // eat heatmap
         cmdstream >> rotation;
 
         if (!cmdstream.fail()) {
-            auto vec = Network::get_scored_moves(
-                &game, Network::Ensemble::DIRECT, rotation, true);
-            Network::show_heatmap(&game, vec, false);
+            if (rotation == "all") {
+                for (int r=0; r<8; r++) {
+                    auto vec = Network::get_scored_moves(
+                        &game, Network::Ensemble::DIRECT, r, true);
+                    Network::show_heatmap(&game, vec, false);
```

Comment:
These two lines are almost the same in all branches, and can be pulled out.

Reply:
I'll see what I can do here. In one branch we need to show 8 heatmaps so I do need that loop. In the other 3 of the 4 branches the show_heatmap command is the same. Each of the 4 branches uses a different get_scored_moves options.

Reply:
I re-worked this. I hope it's what you had in mind. :)